services/app-web/src/pages/CMSDashboard/RateReviewsDashboard/RateReviewsTable.tsx

```diff
@@ -396,22 +396,37 @@ export const RateReviewsTable = ({
     }
 
     const updateRatingPeriodFilter = (
-        date: string | undefined,
-        filterName: 'rateDateStart' | 'rateDateEnd'
+        date: [string | undefined, string | undefined],
+        filterColumn: Column<RateInDashboardType>,
+        elementName: string
     ) => {
-        const filterColumn =
-            filterName === 'rateDateStart'
-                ? rateDateStartColumn
-                : rateDateEndColumn
-
-        lastClickedElement.current = `${filterName}Picker`
+        lastClickedElement.current = elementName
         setTableCaption(null)
 
-        if (!date) {
-            filterColumn.setFilterValue([])
-        } else {
-            filterColumn.setFilterValue([date])
-        }
+        filterColumn.setFilterValue(
+            (value: RatingPeriodFilterType): RatingPeriodFilterType => {
+                const prevDates = value ?? ['', '']
+                const toDate =
+                    date[0] !== undefined
+                        ? date[0]
+                        : prevDates[0] !== undefined
```

Comment:
Suggestion (not blocking):  simplify the logic here and adding comments. Why are we preserving previous dates in the filter here?  I believe `const toDate = date[0] !== undefined ? date[0] : prevDates[0] || ''` is still valid

Reply:
This handles the existing `from` or `to` input date when updating the opposite input. Because this is a single field, the `rating period start`, that we are filtering on, the state has to hold both values in the filter state `[from, to]`. When updating, we need to set either the `to` or `from` in that array while preserving the opposite input value so as to not clear it out. 

It is a little complicated because I kept it as one update function instead of two, one for `to` and one for `from`.

Reply:
I will add a comment.

Reply:
Yes - I like that its one function especially because its acting on the same single data column. A little comment clears this right up about why we preserve the prev values (since inputs are updated one at a time not simultaneously, so unchanged one needs to stay the same).